app/webpacker/components/Results/DataTable.jsx

```diff
@@ -0,0 +1,50 @@
+import { flexRender, getCoreRowModel, useReactTable } from '@tanstack/react-table';
+import { Table } from 'semantic-ui-react';
+import React from 'react';
+
+export default function DataTable({ rows, config }) {
+  const table = useReactTable({
+    data: rows || [],
+    columns: config,
+    getCoreRowModel: getCoreRowModel(),
+  });
+
+  return (
+    <div style={{ overflowX: 'scroll' }}>
+      <Table basic="very" compact="very" striped unstackable singleLine>
+        <Table.Header>
+          {table.getHeaderGroups().map((headerGroup) => (
+            <Table.Row key={headerGroup.id}>
+              {headerGroup.headers.map((header) => (
+                <Table.HeaderCell key={header.id} colSpan={header.column.columnDef.colSpan}>
+                  {header.isPlaceholder
+                    ? null
+                    : flexRender(header.column.columnDef.header, header.getContext())}
+                </Table.HeaderCell>
+              ))}
+            </Table.Row>
+          ))}
+        </Table.Header>
+        <Table.Body>
+          {table.getRowModel().rows.map((row) => (
+            <Table.Row key={row.id}>
+              {row.getVisibleCells().map((cell) => (
+                cell.column.columnDef.isMultiAttemptsHack
+                  ? (
+                    <React.Fragment key={cell.id}>
```

Comment:
I was wondering how this was going to work - and seems the answer is 'not well'.

I also thought this was some kind of exception for mbf for a long time.

Reply:
Hm... I could play around with trying to align it using a sub-header, but if the answer is not forthcoming after ~30 minutes I will just let it be.

Reply:
Hm, it is easy to specify but cumbersome to style. If you have a definition similar to
```js
[
  { accessorKey: 'competitionName' },
  { accessorKey: 'personName' },
  { header: 'Successful solves', columns: [
    { accessorKey: 'value1' },
    { accessorKey: 'value2' },
    { accessorKey: 'value3' },
  ] }
]
```
then Tanstack pushes the competition name and person name header _down_ to be on the same level as the header for `value1`, `value2`, etc.
![image](https://github.com/user-attachments/assets/b0d1780b-5932-4318-80bd-6b2201d21217)

In other words, it always pushes down individual column headings as far down as possible. If we could instead set this to pull them up as far as possible, I can adjust the `rowSpan` to make it look nice.

Reply:
Couldn't find anything in their documentation, so I will go ahead and raise an Issue / discussion

Reply:
Seems there is https://github.com/TanStack/table/discussions/5051 but no official word on whether or when this will be officially supported.

Reply:
Okay, I managed to implement something based on the discussion linked above:
![image](https://github.com/user-attachments/assets/64206090-57a2-40db-853c-419a6b2b9f96)

Reply:
I think the header margins make this look a bit bloated... Also, it's gonna be tricky figuring out which results to wrap

Reply:
I think the way the 'hack' ends up looking is the best, with a single column of span 5. I would leave it as-is then. Possibly renaming `isMultiAttemptsHack` to `rendersOwnCells` or something like that, which explains the fragment replacing the cell.